hazelcast/src/main/java/com/hazelcast/map/impl/record/RecordReaderWriter.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl.record;
+
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.Data;
+
+import java.io.IOException;
+
+/**
+ * Used when reading and writing records
+ * for backup and replication operations
+ */
+public enum RecordReaderWriter {
+    DATA_RECORD_READER_WRITER(TypeId.DATA_RECORD_TYPE_ID) {
+        @Override
+        void writeRecord(ObjectDataOutput out,
+                         Record record, Data dataValue) throws IOException {
+            out.writeByte(record.getMatchingRecordReaderWriterId());
```

Comment:
It's going to get a bit tricky if we need to introduce RU checks, then tracking from the RU check in this class back to the operation which uses this method.